### PR TITLE
TVIST1-555: Add checks for documents being (soft)deletable

### PR DESCRIPTION
**Pull request by jekuaitk**
* Adds check for whether documents may be (soft)deleted
* Translates document types used in `DocumentFilterType`

https://jira.itkdev.dk/browse/TVIST1-555

### CODE REVIEW

File: src/Twig/TwigExtension.php [hunk 1]
```diff
@@ -55,4 +56,41 @@ public function dateNullableFilter($timestamp, $format, $nullDisplayValue = ''):
...
+        foreach ($agendaCaseItems as $agendaCaseItem) {
+            $agendaCaseItemDocuments = $agendaCaseItem->getDocuments();
+            foreach ($agendaCaseItemDocuments as $agendaCaseItemDocument) {
+                if ($document->getId() === $agendaCaseItemDocument->getId()) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
```
Comment (rimi-itk) [inline src/Twig/TwigExtension.php:94]: This logic should be moved to a service. It may come in handy in a controller, say.

**Review by rimi-itk — APPROVED** [review]
Approved with a comment/suggestion.

File: src/Twig/TwigExtension.php [hunk 2]
```diff
@@ -55,4 +56,41 @@ public function dateNullableFilter($timestamp, $format, $nullDisplayValue = ''):
...
+            }
+
+            $attachments = $digitalPost->getAttachments();
+
+            foreach ($attachments as $attachment) {
+                if ($attachment->getDocument()->getId() === $document->getId()) {
+                    return false;
+                }
+            }
+        }
```
Comment (rimi-itk) [inline src/Twig/TwigExtension.php:80]: It must/should be possible to do something like

```php
$items = $someRepository->findDigitalPostAttachmentsByDocument(…);
if (count($items) > 0) {
  return false;
}
```

File: src/Twig/TwigExtension.php [hunk 3]
```diff
@@ -55,4 +56,41 @@ public function dateNullableFilter($timestamp, $format, $nullDisplayValue = ''):
...
+        $agendaCaseItems = $case->getAgendaCaseItems();
+
+        foreach ($agendaCaseItems as $agendaCaseItem) {
+            $agendaCaseItemDocuments = $agendaCaseItem->getDocuments();
+            foreach ($agendaCaseItemDocuments as $agendaCaseItemDocument) {
+                if ($document->getId() === $agendaCaseItemDocument->getId()) {
+                    return false;
+                }
+            }
+        }
```
Comment (rimi-itk) [inline src/Twig/TwigExtension.php:92]: It must/should be possible to do something like

```php
$items = $someRepository->findAgendaCaseItemsByDocument(…);
if (count($items) > 0) {
  return false;
}
```

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: src/Twig/TwigExtension.php (hunk 1 above)
Comment (jekuaitk) [reply src/Twig/TwigExtension.php]: Moved to `DocumentDeletableHelper` service.

File: src/Twig/TwigExtension.php (hunk 2 above)
Comment (jekuaitk) [reply src/Twig/TwigExtension.php]: Yes - updated to use repository methods.

File: src/Twig/TwigExtension.php (hunk 3 above)
Comment (jekuaitk) [reply src/Twig/TwigExtension.php]: Yes - updated to use repository methods.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**